Use system glslang and SPIRV-Cross.
Don't reach for Wayland headers.
libHLSL is gone since glslang 14

Index: gtk/CMakeLists.txt
--- gtk/CMakeLists.txt.orig
+++ gtk/CMakeLists.txt
@@ -85,16 +85,10 @@ list(APPEND INCLUDES ../external/glad/include)
 
 if(USE_SLANG)
     list(APPEND SOURCES ../shaders/slang.cpp)
-    list(APPEND INCLUDES ../external/glslang)
 
     set(SPIRV_CROSS_EXCEPTIONS_TO_ASSERTIONS CACHE BOOL ON)
     set(BUILD_TESTING CACHE BOOL OFF)
-    add_subdirectory("../external/glslang" "glslang" EXCLUDE_FROM_ALL)
-    add_subdirectory("../external/SPIRV-Cross" "SPIRV-Cross" EXCLUDE_FROM_ALL)
     list(APPEND LIBS glslang
-                     OGLCompiler
-                     HLSL
-                     OSDependent
                      SPIRV
                      glslang-default-resource-limits)
     list(APPEND LIBS spirv-cross-core
@@ -102,10 +96,8 @@ if(USE_SLANG)
                      spirv-cross-reflect
                      spirv-cross-cpp)
     list(APPEND DEFINES "USE_SLANG")
-    list(APPEND INCLUDES "../external/glslang")
 
     list(APPEND DEFINES "VK_USE_PLATFORM_XLIB_KHR"
-                        "VK_USE_PLATFORM_WAYLAND_KHR"
                         "VULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1"
                         "VULKAN_HPP_NO_NODISCARD_WARNINGS=1"
                         "VULKAN_HPP_NO_EXCEPTIONS=1"
